### ISSUE TITLE
[Main screen] Top app bar isn't visible sometimes

### ISSUE THREAD

**Issue by min7-i** [breezy-weather/breezy-weather#166]
### Steps to reproduce

In some cases, presumably when weather data is refreshed, the top app bar isn't visible on the main screen. Unfortunately, I haven't found a way to reproduce this reliable, but I had the most success with the following approach:

1. In weather provider settings change the OpenWeather One Call Version to version 3.0 and keep the default API key, so that the weather data refresh will fail.
2. Go back and open your location list.
3. Choose a location via OpenWeather, e.g. Madrid, Spain.
4. Choose a second location via OpenWeather, e.g. Barcelona, Spain.
5. Tap on Barcelona, Spain.

https://github.com/breezy-weather/breezy-weather/assets/97251923/2507bbb9-92d0-41cb-a452-06eafdcd9ae5

### Expected behavior

The main screen is shown and the top app bar, including the location name "Barcelona", is visible.

### Actual behavior

Only the default background is shown on the main screen. The top app bar isn't visible.

### Weather provider used

N/A

### Breezy Weather version

4.2.0-beta

### Android version

Android 13 (GrapheneOS)

### Device

Google Pixel 6a

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to **[latest version](https://github.com/breezy-weather/breezy-weather/releases/latest)**.

**Comment by min7-i** [comment]
Actually, I'm not sure if it's related to the background refresh. Sometimes it also happens when randomly switching between locations via the location list.

https://github.com/breezy-weather/breezy-weather/assets/97251923/f95f9874-0669-4d60-9fd3-e925d3755cf7

**Comment by papjul** [comment]
Your step-by-step no longer reproduces the issue in v4.3.0-beta, however I'm pretty sure I still came accross it recently.

If you can find a reliable (and preferably short) way to get this issue, I will use it to test some fixes and pick the one that works.

**Comment by min7-i** [comment]
I've found a way to reproduce it on my phone:
1. Add the current location.
2. Tap on it and wait on the main screen until weather data for your current location was fetched.
3. Go back to the location list and change the weather provider for your current location (swipe left).
4. Tap on it again and watch the main screen.
=> Now the top app bar shouldn't be visible.
